frontend/src/data/DataFetcher.ts

```diff
@@ -54,29 +66,18 @@ class DataFetcher {
    * The base url for API calls. Empty string if API calls are relative to the
    * current domain.
    */
-  apiUrl: string;
-
-  constructor() {
-    // If the API url isn't provided, requests are relative to current domain.
-    // TODO it might be better to wrap environment variable access in an API
-    // that automatically adds these prefixes and checks both. That also allows
-    // centralizing environment variables so they're easier to audit, fake, etc.
-    this.apiUrl =
-      process.env.REACT_APP_BASE_API_URL ||
-      process.env.STORYBOOK_BASE_API_URL ||
-      "";
-
-    // Use the static file directory for unit test environments, and for
-    // localhost environments unless the API url is provided
-    // TODO we should replace this class with a FakeDataFetcher for test
-    // environments and make this class throw an error in test environments.
-    this.forceStaticFile =
-      process.env.NODE_ENV === "test" ||
-      (process.env.NODE_ENV === "development" && !this.apiUrl);
+  baseApiUrl: string;
+
+  constructor(baseApiUrl: string, deployContext: DeployContext) {
+    this.baseApiUrl = baseApiUrl;
+
+    // Use the static file directory for development environments unless the API
+    // url is provided
+    this.forceStaticFile = deployContext === "development" && !this.baseApiUrl;
```

Comment:
I've noticed Storybook is actually kinda slow (DatasetStore isn't shared between stories) so I'm inclined to have Storybook also read from static file. We could add a TODO to follow up

Reply:
Oh interesting. I think the tricky part about this is in the long-term we don't want to be keeping any static data files in github, which is the only way to deploy these. I don't think there's a way around that unfortunately.